test/functional/shared.js

```diff
@@ -192,12 +192,30 @@ class EventCollector {
   }
 }
 
+function withMonitoredClient(commands, callback) {
+  if (!callback.hasOwnProperty('prototype')) {
+    throw new Error('withMonitoredClient callback can not be arrow function');
+  }
+  return function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ monitorCommands: true });
+    const events = [];
+    client.on('commandStarted', filterForCommands(commands, events));
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const d = () => client.close(done);
```

Comment:
this looks like it would ignore if there was an error in the child code:
```
it('might fail', withMonitoredClient(['find], (client, events, done) => {
   done(new Error('something bad happened!');
});
```

looks like it would just close the client, and report back a successful run. Maybe it makes sense to add some tests for the `withMonitoredClient` helper? Another thing that occurs to me is that it will not presently handle async/Promises:

```
it('should work with promises', withMonitoredClient(['find], function(client, events) {
  return client.insertOne({ a: 42 });
});

it('should work with async/await', withMonitoredClient(['find], async function(client, events) {
  await client.insertOne({ a: 42 });
});
```